File: packages/ui-swiftui/Sources/KinetixUI/Menubar.swift
//
// Menubar.swift — KinetixMenubar / KinetixMenubarMenu.
//
// Mirrors packages/ui/src/components/menubar.tsx: a bordered bar of
// top-level menus. Each KinetixMenubarMenu wraps a native `Menu`; fill
// its content with the shared `KinetixMenuItem` / `KinetixMenuSeparator`
// / `KinetixMenuLabel` parts from DropdownMenu.swift.
//

import SwiftUI

public struct KinetixMenubar<Content: View>: View {
    @Environment(\.kinetixColors) private var colors
    private let content: Content

    public init(@ViewBuilder content: () -> Content) {
        self.content = content()
    }

    public var body: some View {
        HStack(spacing: 4) { content }
            .padding(4)
            .background(colors.background, in: RoundedRectangle(cornerRadius: 8, style: .continuous))
            .overlay {
                RoundedRectangle(cornerRadius: 8, style: .continuous)
                    .strokeBorder(colors.border, lineWidth: 1)
            }
    }
}

public struct KinetixMenubarMenu<Content: View>: View {
    @Environment(\.kinetixColors) private var colors
    private let title: String
    private let menuContent: Content

    public init(_ title: String, @ViewBuilder content: () -> Content) {
        self.title = title
        self.menuContent = content()
    }

    public var body: some View {
        Menu {
            menuContent
        } label: {
            Text(title)
                .font(.system(size: 13, weight: .medium))
                .foregroundStyle(colors.foreground)
                .padding(.horizontal, 8)
                .padding(.vertical, 4)
                .contentShape(Rectangle())
        }
    }
}

File: packages/ui-flutter/lib/src/menubar.dart
import 'package:flutter/material.dart';

import 'theme.dart';

/// Mirrors `packages/ui/src/components/menubar.tsx`: a bordered bar of
/// top-level menus. Wraps Flutter's `MenuBar` / `SubmenuButton`; fill
/// each menu with the shared `KinetixMenuItem` / `KinetixMenuSeparator` /
/// `KinetixMenuLabel` parts from dropdown_menu.dart.
class KinetixMenubar extends StatelessWidget {
  const KinetixMenubar({super.key, required this.children});

  final List<Widget> children;

  @override
  Widget build(BuildContext context) {
    final c = KinetixTheme.of(context);
    return MenuBar(
      style: MenuStyle(
        backgroundColor: WidgetStatePropertyAll(c.background),
        surfaceTintColor: const WidgetStatePropertyAll(Color(0x00000000)),
        elevation: const WidgetStatePropertyAll(0),
        shape: WidgetStatePropertyAll(
          RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(8),
            side: BorderSide(color: c.border),
          ),
        ),
      ),
      children: children,
    );
  }
}

class KinetixMenubarMenu extends StatelessWidget {
  const KinetixMenubarMenu(this.title, {super.key, required this.menuChildren});

  final String title;
  final List<Widget> menuChildren;

  @override
  Widget build(BuildContext context) {
    final c = KinetixTheme.of(context);
    return SubmenuButton(
      menuChildren: menuChildren,
      child: Text(
        title,
        style: TextStyle(fontSize: 13, fontWeight: FontWeight.w500, color: c.foreground),
      ),
    );
  }
}
